api/resourceserver/check_webhook.go

```diff
@@ -0,0 +1,69 @@
+package resourceserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc"
+	"github.com/concourse/atc/db"
+	"github.com/concourse/atc/dbng"
+	"github.com/concourse/atc/resource"
+	"github.com/tedsuo/rata"
+)
+
+// CheckResourceWebHook defines a handler for process a check resource request via an access token.
+func (s *Server) CheckResourceWebHook(pipelineDB db.PipelineDB, dbPipeline dbng.Pipeline) http.Handler {
+	logger := s.logger.Session("check-resource-webhook")
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resourceName := rata.Param(r, "resource_name")
+		accessToken := r.URL.Query().Get("access_token")
+
+		if accessToken == "" {
+			logger.Info("malformed-request", lager.Data{"error": "missing access_token"})
```

Comment:
I wouldn't lean on the existing BasicAuthValidator as that is specifically for team-configured basic auth. This is sort of its own way of checking auth, particular to this endpoint, so it could just interpret the `Authorization` header itself, with it having been ignored by all the other middleware.

Maybe it could just statically check for `webhook` as the user and the configured password as the password?